aws_signing_helper/signer.go

```diff
@@ -205,35 +197,22 @@ func GetSigner(opts *CredentialsOpts) (signer Signer, signatureAlgorithm string,
 					" within the PKCS#12 file")
 			}
 			// Not a PEM certificate? Try PKCS#12
-			certificateChain, privateKey, err = ReadPKCS12Data(opts.CertificateId)
+			certificateChain, _, err = ReadPKCS12Data(opts.CertificateId)
 			if err != nil {
 				return nil, "", err
 			}
-			if privateKey != nil {
-				ecPrivateKeyPtr, isEcKey := privateKey.(*ecdsa.PrivateKey)
-				if isEcKey {
-					privateKey = *ecPrivateKeyPtr
-				}
-
-				rsaPrivateKeyPtr, isRsaKey := privateKey.(*rsa.PrivateKey)
-				if isRsaKey {
-					privateKey = *rsaPrivateKeyPtr
-				}
-			}
-			return GetFileSystemSigner(privateKey, certificateChain[0], certificateChain)
+			return GetFileSystemSigner(opts.PrivateKeyId, opts.CertificateId, opts.CertificateBundleId, true)
```

Comment:
If we're using PKCS#12 here, I think previous assumptions have been that the certificate bundle would be stored in the PKCS#12 container. Not sure if making this change hurts though... What happens in the case that there's a PKCS#12 bundle file referenced with a certificate bundle in it as well as a "normal" certificate bundle file? Will the contents of the two bundles add to each other?

Reply:
I'm not really sure I follow.  The previous implementation was calling `ReadPKCS12Data` and passing what was read to a file system signer to be cached for the duration of the process.  The new implementation is still calling `ReadPKCS12Data` and if it doesn't get an error back then it passes the same path information that was available here to the new file system signer so that it can call `ReadPKCS12Data` when it needs to read the certs if `isPkcs12` is true.

Reply:
The previous implementation ignores `opts.CertificateBundleId` when `opts.CertificateId` contains a path to a PKCS#12 file. In the PKCS#12 case, we expect the certificate bundle to be provided in the PKCS#12 container specified by `opts.CertificateId`, so technically, `opts.CertificateBundleId` isn't used at all I believe. In your changes, you're passing in `opts.CertificateBundleId` into `GetFileSystemSigner` in order to create the `FileSystemSigner`. It's possible that they're functionally equivalent, but I wanted to make sure.

Reply:
When `isPkcs12` is true and the `FileSystemSigner` reads the cert data it ignores `opts.PrivateKeyId` and `opts.CertificateBundleId`
https://github.com/aws/rolesanywhere-credential-helper/pull/65/files#diff-9760200f6d4074d48601f0b9219340d8f7706d99fc9066d45527e99eabcbecc7R112-R119